### PR TITLE
New `quotes_linter()` to replace `single_quotes_linter()`

### LINKED ISSUE #1729: quotes_linter()
I think it makes sense to offer a generalized version of `single_quotes_linter()` that allows users to choose `'` as their preferred quoting character. Of course `"` should be the default.

### PR DESCRIPTION
**Pull request by MichaelChirico**
Closes #1729

### CODE REVIEW

File: R/quotes_linter.R [hunk 1]
```diff
@@ -0,0 +1,84 @@
+#' Character string quote linter
+#'
+#' Check that the desired quoting character is used to delimit string constants.
+#'
+#' @param quote_char Which quote character to accept. Defaults to the tidyverse
```
Comment (IndrajeetPatil) [inline R/quotes_linter.R:5]: I am wondering if `delimiter` might be a better parameter name here than `quote_char`? 🤔

Comment (MichaelChirico) [reply R/quotes_linter.R:5]: good idea -- that's the term used in ?Quotes

screen shot of ?Quotes text esp. "Single and double quotes delimit character constants."

Comment (MichaelChirico) [reply R/quotes_linter.R]: Done

File: NEWS.md [hunk 2]
```diff
@@ -131,6 +136,11 @@
 
 * `implicit_assignment_linter()` for checking implicit assignments in function calls (@IndrajeetPatil and @AshesITR, #1777).
 
+* `quotes_linter()` is a generalized version of (now deprecated) `single_quotes_linter()`. It accepts an argument `delimite_char` to specify whether `"` or `'` should be the accepted method for delimiting character literals. The default, `"`, reflects the Tidyverse guide recommendation and matches the behavior of `single_quotes_linter()`.
+* `quotes_linter()` is a generalized version of (now deprecated) `single_quotes_linter()`. It accepts an argument `quote_char` to specify whether `"` or `'` should be the accepted method for quoting character literals. The default, `"`, reflects the Tidyverse style guide recommendation and matches the behavior of `single_quotes_linter()`.
```
Comment (IndrajeetPatil) [inline NEWS.md:140]:
```suggestion
```

File: R/quotes_linter.R [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    quote_regex <- rex(
+      start,
+      zero_or_one(character_class("rR")),
+      double_quote,
+      any_non_single_quotes,
+      double_quote,
+      end
+    )
+    lint_message <- "Only use single-quotes."
+  }
```
Comment (IndrajeetPatil) [inline R/quotes_linter.R:61]:
```suggestion
  }
  
```

File: NEWS.md (hunk 2 above)
Comment (MichaelChirico) [reply NEWS.md:140]: not sure what happened there

**Review by IndrajeetPatil — APPROVED** [review]

**Pull request merged**